use-cases.md

```diff
@@ -163,37 +146,42 @@ When an application acts as an RPC client, it is expected to start a new tracing
-  * We assume the HTTP client is asynchronous, so it returns a Future, and we need to add an on-completion callback to be able to finish the current child Span.
-  * If the HTTP client returns a future with exception, we log the exception to the Span with `log_event_with_payload` method.
+Notes on this example:
+  * The `get_current_context()` function is not a part of the OpenTracing API. It is meant to represent some util method of retrieving the current TraceContext, which can be achieved either implicitly or explicitly as discussed above.
+  * We assume the HTTP client is asynchronous, so it returns a Future, and we need to add an on-completion callback to be able to finish the current child Span.  The synchronous case is trivially similar for instrumentation.
+  * If the HTTP client returns a future with exception, we log the exception to the Span with `log` method.
   * Because the HTTP client may throw an exception even before returning a Future, we use a try/catch block to finish the Span in all circumstances, to ensure it is reported and avoid leaking resources.
 
 
-### Using Distributed Context Propagation
+### Using Trace Attributes
 
-The client and server examples above propagated the Span/Trace over the wire, including any Trace Attributes. The client may use the Trace Attributes to pass additional data to the server and any other downstream server it might call.
+The client and server examples above propagated the TraceContext over the wire, including both Tracer-specific state related to the current request as well as user-added Trace Attributes. The client may use the Trace Attributes to pass additional data to the server and any other downstream server it might call.
 
 ```python
 
     # client side
-    span.set_trace_attribute('auth-token', '.....')
+    ctx.set_attribute('auth-token', '.....')
+
+    ...
 
     # server side (one or more levels down from the client)
-    token = span.get_trace_attribute('auth-token')
+    token = ctx.get_attribute('auth-token')
 ```
 
 ### Logging Events
 
-We have already used `log_event_with_payload` in the client Span use case. Events can be logged without a payload, and not just where the Span is being created / finished. For example, the application may record a cache miss event in the middle of execution, as long as it can get access to the current Span from the request context:
+We have already used `log` in the client Span use case. Events can be logged without a payload, and not just where the Span is being created / finished. For example, the application may record a cache miss event in the middle of execution, as long as it can get access to the current Span:
 
 ```python
 
-    span = get_current_span()
-    span.log_event('cache-miss') 
+    # if the Span isn't handy, grab it
+    span = get_current_context().get_current_span()
```

Comment:
however, conceptually it makes the following very suspicious:
```
span = tracer.create_span(operation_name=operation, context=ctx)
```
Maybe this should be `tracer.create_context`, since we do need to create a child context with a child span inside it. Also, returning a `span` directly instead of the child context does not allow for in-process propagation of that context (unless we assume in-process propagation is also built-in into OT, which we didn't want - won't work in Go anyway).